server/hydrogen-render/1-render-hydrogen-to-string.js

```diff
@@ -24,36 +33,56 @@ async function renderHydrogenToString(options) {
     // we receive the SSR results.
     const child = fork(require.resolve('./2-render-hydrogen-to-string-fork-script'), [], {
       signal,
+      // Default to silencing logs from the child process. We already have
+      // proper instrumentation of any errors that might occur.
+      //
+      // This also makes `child.stderr` and `child.stdout` available
+      silent: true,
       //cwd: process.cwd(),
     });
 
-    // Pass the options to the child by sending instead of via argv because we
+    // Since we have to use the `silent` option for the `stderr` stuff below, we
+    // should also print out the `stdout` to our main console.
+    if (logOutputFromChildProcesses) {
+      child.stdout.on('data', function (data) {
+        console.log('Child printed something to stdout:', String(data));
+      });
+
+      child.stderr.on('data', function (data) {
+        console.log('Child printed something to stderr:', String(data));
+      });
+    }
+
+    // Pass the renderOptions to the child by sending instead of via argv because we
     // will run into `Error: spawn E2BIG` and `Error: spawn ENAMETOOLONG` with
     // argv.
-    child.send(options);
+    child.send(renderOptions);
 
     // Stops the child process if it takes too long
     setTimeout(() => {
       controller.abort();
     }, RENDER_TIMEOUT);
```

Comment:
Currently, the CI is failing by failing to render Hydrogen in time:

https://github.com/matrix-org/matrix-public-archive/runs/7460940355?check_suite_focus=true
```
RethrownError: Failed to render Hydrogen to string. In order to reproduce, feed in these arguments into `renderHydrogenToString(...)`:
    renderToString arguments: {"fromTimestamp":1658448000000,"roomData":{"id":"!ohgbbOvmGMOCMRyyej:hs1","name":"the hangout spot"},"events":[{"type":"m.room.create","room_id":"!ohgbbOvmGMOCMRyyej:hs1","sender":"@user-t1658454780380-r828377575:hs1","content":{"room_version":"9","creator":"@user-t1658454780380-r828377575:hs1"},"state_key":"","origin_server_ts":1658454780402,"unsigned":{"age":253},"event_id":"$DhohahyOKvAuZYaep2R55p2Kwo4GxzKOa9FFbJ0ZrOw","user_id":"@user-t1658454780380-r828377575:hs1","age":253},{"type":"m.room.member","room_id":"!ohgbbOvmGMOCMRyyej:hs1","sender":"@user-t1658454780380-r828377575:hs1","content":{"membership":"join","displayname":"user-t1658454780380-r828377575"},"state_key":"@user-t1658454780380-r828377575:hs1","origin_server_ts":1658454780434,"unsigned":{"age":221},"event_id":"$Q1E_NqZFP1CoKwEQ4O3hE2BcyjokEk7K-RA_mabekKk","user_id":"@user-t1658454780380-r828377575:hs1","age":221},{"type":"m.room.power_levels","room_id":"!ohgbbOvmGMOCMRyyej:hs1","sender":"@user-t1658454780380-r828377575:hs1","content":{"users":{"@user-t1658454780380-r828377575:hs1":100},"users_default":0,"events":{"m.room.name":50,"m.room.power_levels":100,"m.room.history_visibility":100,"m.room.canonical_alias":50,"m.room.avatar":50,"m.room.tombstone":100,"m.room.server_acl":100,"m.room.encryption":100},"events_default":0,"state_default":50,"ban":50,"kick":50,"redact":50,"invite":50,"historical":100},"state_key":"","origin_server_ts":1658454780457,"unsigned":{"age":198},"event_id":"$TZxBK83_CyDR6HuLe5Q4yoXS8CFgfyzB-KaFCh-SVYg","user_id":"@user-t1658454780380-r828377575:hs1","age":198},{"type":"m.room.join_rules","room_id":"!ohgbbOvmGMOCMRyyej:hs1","sender":"@user-t1658454780380-r828377575:hs1","content":{"join_rule":"public"},"state_key":"","origin_server_ts":1658454780498,"unsigned":{"age":157},"event_id":"$9XRxxvX7N-rSKPKLBxv1dl9WB8odGz-OKRVOMmIsj8A","user_id":"@user-t1658454780380-r828377575:hs1","age":157},{"type":"m.room.history_visibility","room_id":"!ohgbbOvmGMOCMRyyej:hs1","sender":"@user-t1658454780380-r828377575:hs1","content":{"history_visibility":"world_readable"},"state_key":"","origin_server_ts":1658454780526,"unsigned":{"age":129},"event_id":"$QYA35VbwdrbuQEmgyrC0jfYIUu8iAaKbgtIUF11vQig","user_id":"@user-t1658454780380-r828377575:hs1","age":129},{"type":"m.room.name","room_id":"!ohgbbOvmGMOCMRyyej:hs1","sender":"@user-t1658454780380-r828377575:hs1","content":{"name":"the hangout spot"},"state_key":"","origin_server_ts":1658454780559,"unsigned":{"age":96},"event_id":"$_0M33axNTXyVOoYJhkg9ESsID2DB232Zwl0jW8LPZ3c","user_id":"@user-t1658454780380-r828377575:hs1","age":96},{"type":"m.room.member","room_id":"!ohgbbOvmGMOCMRyyej:hs1","sender":"@archiver:hs1","content":{"membership":"join"},"state_key":"@archiver:hs1","origin_server_ts":1658454780603,"unsigned":{"age":52},"event_id":"$D_an4-oheIM4oNYxaBuSPh8z7rNsyEUKfOQ2OoK3hbg","user_id":"@archiver:hs1","age":52}],"stateEventMap":{"@user-t1658454780380-r828377575:hs1":{"type":"m.room.member","room_id":"!ohgbbOvmGMOCMRyyej:hs1","sender":"@user-t1658454780380-r828377575:hs1","content":{"membership":"join","displayname":"user-t1658454780380-r828377575"},"state_key":"@user-t1658454780380-r828377575:hs1","origin_server_ts":1658454780434,"unsigned":{"age":221},"event_id":"$Q1E_NqZFP1CoKwEQ4O3hE2BcyjokEk7K-RA_mabekKk","user_id":"@user-t1658454780380-r828377575:hs1","age":221},"@archiver:hs1":{"type":"m.room.member","room_id":"!ohgbbOvmGMOCMRyyej:hs1","sender":"@archiver:hs1","content":{"membership":"join"},"state_key":"@archiver:hs1","origin_server_ts":1658454780603,"unsigned":{"age":52},"event_id":"$D_an4-oheIM4oNYxaBuSPh8z7rNsyEUKfOQ2OoK3hbg","user_id":"@archiver:hs1","age":52}}}
    at renderHydrogenToString (/home/runner/work/matrix-public-archive/matrix-public-archive/server/hydrogen-render/1-render-hydrogen-to-string.js:135:11)
    --- Original Error ---
    RethrownError: Timed out while rendering Hydrogen to string so we aborted the child process after 5000ms
        at ChildProcess.<anonymous> (/home/runner/work/matrix-public-archive/matrix-public-archive/server/hydrogen-render/1-render-hydrogen-to-string.js:118:17)
        --- Original Error ---
        AbortError: The operation was aborted
            at abortChildProcess (node:child_process:663:27)
            at AbortSignal.onAbortListener (node:child_process:733:7)
            at AbortSignal.[nodejs.internal.kHybridDispatch] (node:internal/event_target:643:20)
            at AbortSignal.dispatchEvent (node:internal/event_target:585:26)
            at abortSignal (node:internal/abort_controller:284:10)
            at AbortController.abort (node:internal/abort_controller:315:5)
            at Timeout._onTimeout (/home/runner/work/matrix-public-archive/matrix-public-archive/server/hydrogen-render/1-render-hydrogen-to-string.js:63:18)
            at listOnTimeout (node:internal/timers:559:17)
            at processTimers (node:internal/timers:502:7)
```

Reply:
Must be something with `fake-indexeddb` or the mock `LocalStorage` because the tests pass after removing it https://github.com/matrix-org/matrix-public-archive/pull/43/commits/74b096c1fcdd9ba7889d49d96ea12bb4b9a5c0a0